base/compiler/abstractinterpretation.jl

```diff
@@ -1858,7 +1858,12 @@ function abstract_call_unionall(interp::AbstractInterpreter, argtypes::Vector{An
         a2 = argtypes[2]
         a3 = argtypes[3]
         ⊑ᵢ = ⊑(typeinf_lattice(interp))
-        nothrow = a2 ⊑ᵢ TypeVar && (a3 ⊑ᵢ Type || a3 ⊑ᵢ TypeVar)
+        if isvarargtype(a3)
```

Comment:
The `length(argtypes)` check is not correct either if you have varargs

Reply:
Yeah, but what we are trying to model here is call to the `UnionAll(tv, t)` method, which just throws `MethodError` when `length(argtypes) ≠ 3` at runtime. So this is fine as far as we model `nothrow` correctly.

Reply:
When you have Vararg, you don’t know if length!=3, so you don’t know that if it throws an error or not

Reply:
Yes, we can't concluded that the rt is `Bottom` if either of the first two argtypes is `Vararg`, we should fix that.

Reply:
Ah, right... I will work on it.